servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -240,21 +230,26 @@ private static void updateResponseContentLength(final int contentLength, final H
         });
     }
 
-    static StreamingHttpResponse addResponseTransferEncodingIfNecessary(final StreamingHttpResponse response,
-                                                                        final HttpRequestMethod requestMethod) {
-        if (canAddResponseTransferEncoding(response, requestMethod)) {
+    static void addResponseTransferEncodingIfNecessary(final StreamingHttpResponse response,
+                                                       final HttpRequestMethod requestMethod) {
+        if (canAddResponseTransferEncodingProtocol(response.status().code(), requestMethod) &&
+                canAddTransferEncodingChunked(response)) {
             response.headers().add(TRANSFER_ENCODING, CHUNKED);
         }
-        return response;
     }
 
     static void addRequestTransferEncodingIfNecessary(final StreamingHttpRequest request) {
-        if (canAddRequestTransferEncoding(request)) {
+        if (clientMaySendPayloadBodyFor(request.method()) && canAddTransferEncodingChunked(request)) {
```

Comment:
it is unexpected that the outer method doesn't have `client` in the name, and we are calling a `client` specific method. as discussed offline there are so many methods in this class which do very similar things (e.g. content-length, transfer-encoding) it would be good to clarify in a followup PR.